src/qibolab/instruments/zhinst.py

```diff
@@ -0,0 +1,1050 @@
+        """Add sweeper to list of sweepers"""
+        self.zhsweepers.append(self.select_sweeper(sweeper, qubit))
+
+
+class ZhSweeperLine:
+    """
+    Zurich sweeper from qibolab sweeper for non pulse parameters
+    Bias, Delay (, power_range, local_oscillator frequency, offset ???)
+
+    For now Parameter.bias sweepers are implemented as Parameter.Amplitude
+    on a flux pulse. We may want to keep this class separate for future
+    Near Time sweeps
+    """
+
+    def __init__(self, sweeper, qubit=None, sequence=None):
+        """Qibolab sweeper"""
+        self.sweeper = sweeper
+
+        # TODO: I already created a flux pulse check
+        if sweeper.parameter is Parameter.bias:
+            pulse = FluxPulse(
+                start=sequence.start,
+                duration=sequence.duration,
+                amplitude=1,
+                shape="Rectangular",
+                channel=qubit.flux.name,
+                qubit=qubit.name,
+            )
+            self.pulse = pulse
+            self.signal = f"flux{qubit.name}"
+
+            self.zhpulse = lo.pulse_library.const(
+                uid=(f"{pulse.type.name.lower()}_{pulse.qubit}_"),
+                length=round(pulse.duration * 1e-9, 9),
+                amplitude=pulse.amplitude,
+            )
+
+        # Need something better to store multiple sweeps on the same pulse
+        self.zhsweeper = self.select_sweeper(sweeper)
+
+    def select_sweeper(self, sweeper):
+        """Sweeper translation"""
+        if sweeper.parameter is Parameter.bias:
+            zh_sweeper = lo.SweepParameter(
+                uid=sweeper.parameter.name,
+                values=sweeper.values,
+            )
+        if sweeper.parameter is Parameter.delay:
+            zh_sweeper = lo.SweepParameter(
+                uid=sweeper.parameter.name,
+                values=sweeper.values * 1e-9,
+            )
+        return zh_sweeper
+
+
+class Zurich(AbstractInstrument):
+    """Zurich driver main class"""
+
+    def __init__(self, name, descriptor, use_emulation=False):
+        "Setup name (str)"
+        self.name = name
+        """
+        Port and device mapping in yaml text (str)
+
+        It should be used as a template by adding extra lines for each of the different
+        frequency pulses played thought the same port after parsing the sequence.
+        """
+        self.descriptor = descriptor
+        "Enable emulation mode (bool)"
+        self.emulation = use_emulation
+
+        "Is the device connected ? (bool)"
+        self.is_connected = False
+
+        "Signals to lines mapping"
+        self.signal_map = {}
+        "Zurich calibration object)"
+        self.calibration = lo.Calibration()
+
+        "Zurich device parameters for connection"
+        self.device_setup = None
+        self.session = None
+        self.device = None
+
+        "Parameters read from the runcard not part of ExecutionParameters"
+        self.Fast_reset = False
```

Comment:
Can you use consistent case?
```suggestion
        self.fast_reset = False
```

Reply:
I have a function called fast_reset already, I wiil change the name for this

Reply:
actually this is useless now, if I make it come from ExecutionParameters